lib/internal/test_runner/reporter/tap.js

```diff
@@ -209,7 +219,11 @@ function jsToYaml(indent, name, value) {
     // If the ERR_TEST_FAILURE came from an error provided by user code,
     // then try to unwrap the original error message and stack.
     if (code === 'ERR_TEST_FAILURE' && kUnwrapErrors.has(failureType)) {
-      errStack = cause?.stack ?? errStack;
+      if (cause?.cause) {
```

Comment:
```suggestion
      if (cause != null && ObjectPrototypeHasOwnProperty(cause, 'cause')) {
```
Otherwise it wouldn't detect a falsy cause (e.g. `null`, `undefined`, `false`, empty string, etc.)